src/config.rs

```diff
@@ -91,6 +103,13 @@ impl Config {
             ..Default::default()
         }
     }
+    
+    /// Creates a new temp configuration which will delete the folder upon drop.
+    pub fn new_temp() -> Self {
```

Comment:
I think it would be better to change the syntax to something like `Config::new(path).temporary()`, so that a custom path can still be easily chosen to be temporary.

That would allow something like:

```rs
let path = tempfile::tempdir()?.into_path();
let keyspace = Config::new(path).temporary().open()?;
```